components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreManagementService.java

```diff
@@ -0,0 +1,98 @@
+ */
+
+package org.wso2.carbon.security.keystore;
+
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This service contains the methods to manage certificates of the keystore and client truststore.
+ */
+public interface KeyStoreManagementService {
+
+    /**
+     * Retrieves the list of certificate aliases from the keystore.
+     *
+     * @param filter used to filter the result. Supports sw, ew, eq & co. eg:filter=alias+sw+wso2.
+     * @return the {@link List} of alias.
+     * @throws KeyStoreManagementServerException when retrieving the certificate aliases failed.
+     * @throws KeyStoreManagementClientException when retrieving the certificate aliases failed due to a client error.
+     */
+    List<String> getKeyStoreCertificateAliases(String filter)
+            throws KeyStoreManagementServerException, KeyStoreManagementClientException;
+
+    /**
+     * Retrieves the public certificate from the keystore.
+     *
+     * @return a {@link Map} with public key alias and {@link X509Certificate}.
+     * @throws KeyStoreManagementServerException when retrieving the public certificate failed due to a server error.
+     */
+    Map<String, X509Certificate> getPublicCertificate() throws KeyStoreManagementServerException;
+
+    /**
+     * Retrieves the certificate of the given alias from the keystore.
+     *
+     * @param alias of the certificate.
+     * @return the {@link X509Certificate}
+     * @throws KeyStoreManagementServerException when retrieving the certificate failed due to a server error.
+     * @throws KeyStoreManagementClientException when retrieving the certificate failed due to a client error.
+     */
+    X509Certificate getKeyStoreCertificate(String alias) throws KeyStoreManagementServerException,
+            KeyStoreManagementClientException;
+
+    /**
+     * Retrieves the list of certificate aliases from the client truststore.
+     *
+     * @param filter used to filter the result. Supports sw, ew, eq & co. eg:filter=alias+sw+wso2.
+     * @return the {@link List} of alias
+     * @throws KeyStoreManagementServerException when retrieving the certificate aliases failed due to a server error.
+     * @throws KeyStoreManagementClientException when retrieving the certificate aliases failed due to a client error.
+     */
+    List<String> getClientTrustStoreCertificateAliases(String filter)
+            throws KeyStoreManagementServerException, KeyStoreManagementClientException;
+
+    /**
+     * Retrieves the certificate of the given alias from the client truststore.
+     *
+     * @param alias of the certificate.
+     * @return the {@link X509Certificate}
+     * @throws KeyStoreManagementServerException when retrieving the certificate failed due to a server error.
+     * @throws KeyStoreManagementClientException when retrieving the certificate failed due to a server error.
+     */
+    X509Certificate getClientTrustStoreCertificate(String alias)
```

Comment:
Should we have TrustStore in the method name?

Reply:
fixed